homeassistant/components/device_tracker/__init__.py

```diff
@@ -561,6 +569,28 @@ def get_vendor_for_mac(self):
             if resp is not None:
                 yield from resp.release()
 
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Called when entity about to be added to hass."""
+        state = yield from async_get_last_state(self.hass, self.entity_id)
+        _LOGGER.debug("add entity %s %s", self.entity_id, state)
+        if not state:
+            return
+        self._state = state.state
+
+        _LOGGER.debug(dir(state))
+
+        for attr, var in (
+                (ATTR_SOURCE_TYPE, 'source_type'),
+                (ATTR_GPS_ACCURACY, 'gps_accuracy'),
+                (ATTR_BATTERY, 'source_type'),
+        ):
+            if attr in state.attr:
+                setattr(self, var, state.attr[attr])
+
+        if ATTR_LONGITUDE in state.attr:
+            self.gps = (state.attr[ATTR_LATITUDE], state.attr[ATTR_LATITUDE])
```

Comment:
Indeed, thanks :-)

I still need to do tests that is why I marked it wip. Would prefer to have a helper from input_select